homeassistant/components/influxdb/sensor.py

```diff
@@ -123,6 +130,13 @@ def validate_query_format_for_version(conf: dict) -> dict:
             vol.Required(CONF_WHERE): cv.template,
         }
     ),
+    LANGUAGE_INFLUXQL_RAW: _QUERY_SENSOR_SCHEMA.extend(
+        {
+            vol.Optional(CONF_DB_NAME): cv.string,
+            vol.Required(CONF_QUERY): cv.template,
+            vol.Required(CONF_FIELD): cv.string,
```

Comment:
```suggestion
            vol.Optional(CONF_FIELD, default=DEFAULT_FIELD): cv.string,
```
Think we should use the same default value here. That keeps it consistent. Also making this field optional allows this to be used for V2 Flux queries since specifying a field is something that only makes sense with InfluxQL queries. When talking to V2 systems `field` and `db_name` would just be ignored.

Reply:
TBH you could probably just omit this from the config entirely. Just instruct users to put `as value` in their select statement next to whatever they are selecting to give it the expected name.

Reply:
I thought it was more flexible and robust to set the name of the result of the query as a paraeter of configuration wether using the default 'value' which has in this case to be taken into account in the query.
And it may be simpler to extend to V2 Flux queries as I have in mind that default value for V2 is '_value'?

Reply:
In V2 flux queries it is always `_value`. There's not an option. It's not SQL-like like it is with InfluxQL. Each each attribute of each entity ends up in its own table where every row has a `_time` column specifying the time that row was created and a `_value` column specifying the value at that time. The only other columns are what you designate as keys (tags in the config).

To show you what I mean, here's an example entity in my Home Assistant:
![Screen Shot 2022-01-05 at 2 12 11 PM](https://user-images.githubusercontent.com/2037026/148275081-9fa5e2ba-433d-4556-8dda-be95cc992b8f.png)

Here's an example flux query on my data about this entity as Influx sees it:
```
from(bucket: "michael.degatano's Bucket")
  |> range(start: -1h, stop: now())
  |> filter(fn: (r) => r.domain == "sensor" and r.entity_id == "family_room_couch_outlet")
```
And here's the result:
![Screen Shot 2022-01-05 at 2 10 13 PM](https://user-images.githubusercontent.com/2037026/148274916-0d50e5ed-ca0d-4d69-a0d0-a638a5fb4ad4.png)

Notice how in the left there is a separate table for each attribute you see in that entity. And then within each row of those tables there's really only one column that matters - `_value`. Everything else is just things that make up the key of that entity and the same in every row or a timestamp dating the row.

That's why it has to be optional for Influx2, there is no reason to specify a field here. The field you want will always be `_value`.

Reply:
Thanks for those precisions and explanations about Influx2: no need so to get a field name for the result, just have to keep the `INFLUX_CONF_VALUE_V2`...

Reply:
I saw that with InfluxV2 columns can possibly be renamed with the rename function (`rename(columns: {_value: "result"})` for instance).
Probably very unsual in the context, but the ability to specify a name for the result of the query seems to be the most flexible even in Influx2 mode.
The best solution seems to be to keep this parameter optional in both mode with default value as `_value` (which match the Influx2 default). For InfluxQL, user must specify either the name of the result in the query (and so just have to specify the default configuration value `_value` in the query for the result) or specify the name used in the query (and so specify a `field` in the configuration).
That's how it is implemented in the e7dfb9f commit.